src/ui/libs/DatalensChartkit/ChartKit/components/Widget/components/Markdown/Markdown.tsx

```diff
@@ -15,15 +15,25 @@ import './Markdown.scss';
 
 const b = block('chartkit-markdown');
 
-export function Markdown({data, onLoad, id}) {
-    const generatedId = React.useMemo(() => `${id}_${getRandomCKId()}`, [data, id]);
+type MarkdownProps = Omit<WidgetProps, 'data'> & {
+    data: MarkdownWidget & {
+        data: {
+            markdown?: string;
+            meta?: object;
+        };
+    };
+};
+
+export function Markdown({data, onLoad, id}: MarkdownProps) {
+    const generatedId = React.useMemo(() => `${id}_${getRandomCKId()}`, [id]);
```

Comment:
Why to get rid of data in deps? It seems that it was added on purpose earlier -  to regenerate the id when data changes